src/AbstractOperations/conditional_operations.jl

```diff
@@ -19,6 +19,7 @@ struct ConditionalOperation{LX, LY, LZ, F, C, O, G, M, T} <: AbstractOperation{L
         if func === Base.identity
             func = nothing
         end
+        condition = validate_condition(condition, operand)
```

Comment:
does this have a cost? sometimes it's better to keep the inner constructors clean. Fore xample can this run on GPU?

Reply:
Basically we want to have some simple way of building an object that does not have errors and prints and whatnot embedded in it

Reply:
Hm... I thought it'd be cleaner that way.
But I see your point.